File: app/src/main/java/id/ac/unej/inaparadise/ui/IntroActivity.java
package id.ac.unej.inaparadise.ui;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.text.Html;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

import id.ac.unej.inaparadise.R;
import id.ac.unej.inaparadise.adapter.SliderAdapter;

public class IntroActivity extends AppCompatActivity {

    private ViewPager viewpager;
    private LinearLayout liner;
    private SliderAdapter myadapter;

    private TextView[] mdots;
    private Button next, back, mulai;

    private int mCureentPage;
    ViewPager.OnPageChangeListener viewlistener = new ViewPager.OnPageChangeListener() {
        @Override
        public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

        }

        @Override
        public void onPageSelected(int position) {

            adddots(position);
            mCureentPage = position;

            if (position == 0) {
                next.setEnabled(true);
                back.setEnabled(false);
                back.setVisibility(View.INVISIBLE);

                next.setText("lanjut");
                back.setText("");
            } else if (position == mdots.length - 1) {
                next.setEnabled(true);
                back.setEnabled(true);
                back.setVisibility(View.VISIBLE);

                next.setText("");
                back.setText("kembali");
                setMulai(true);
            } else {
                next.setEnabled(true);
                back.setEnabled(true);
                back.setVisibility(View.VISIBLE);

                next.setText("lanjut");
                back.setText("kembali");
                setMulai(false);
            }

        }

        @Override
        public void onPageScrollStateChanged(int state) {

        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_intro);

        viewpager = (ViewPager) findViewById(R.id.viewpager);
        liner = (LinearLayout) findViewById(R.id.dots);

        next = (Button) findViewById(R.id.nextBtn);
        back = (Button) findViewById(R.id.backBtn);
        mulai = findViewById(R.id.mulai);

        myadapter = new SliderAdapter(this);
        viewpager.setAdapter(myadapter);
        adddots(0);

        viewpager.addOnPageChangeListener(viewlistener);

        next.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                viewpager.setCurrentItem(mCureentPage + 1);
            }
        });

        back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                viewpager.setCurrentItem(mCureentPage - 1);
            }
        });
        mulai.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(IntroActivity.this, LoginActivity.class));
            }
        });
    }

    public void adddots(int i) {

        mdots = new TextView[4];
        liner.removeAllViews();

        for (int x = 0; x < mdots.length; x++) {
            mdots[x] = new TextView(this);
            mdots[x].setText(Html.fromHtml("&#8226;"));
            mdots[x].setTextSize(35);
            mdots[x].setTextColor(getResources().getColor(R.color.gray));

            liner.addView(mdots[x]);
        }
        if (mdots.length > 0) {
            mdots[i].setTextColor(getResources().getColor(R.color.colorPrimary));
        }

    }

    public void setMulai(boolean status) {
        if (status) {
            mulai.setVisibility(View.VISIBLE);
        } else {
            mulai.setVisibility(View.GONE);
        }
    }

}

File: app/src/main/java/id/ac/unej/inaparadise/ui/IntroActivity.kt
package id.ac.unej.inaparadise.ui

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import id.ac.unej.inaparadise.R
import android.support.design.widget.BottomNavigationView
import id.ac.unej.inaparadise.fragment.Upload
import id.ac.unej.inaparadise.fragment.Notifikasi
import android.content.Intent
import id.ac.unej.inaparadise.ui.CommunityActivity
import android.support.design.widget.TabLayout
import android.support.v4.view.ViewPager
import id.ac.unej.inaparadise.fragment.TabKebudayaan
import id.ac.unej.inaparadise.fragment.TabPermainan
import id.ac.unej.inaparadise.fragment.TabInformasi
import android.support.design.widget.CoordinatorLayout
import id.ac.unej.inaparadise.adapter.SliderAdapter
import android.support.v4.view.ViewPager.OnPageChangeListener
import id.ac.unej.inaparadise.ui.LoginActivity
import android.text.Html
import android.view.View
import android.widget.*
import id.ac.unej.inaparadise.ui.MainActivity
import id.ac.unej.inaparadise.fragment.TabPesan
import id.ac.unej.inaparadise.fragment.TabKomunitas
import id.ac.unej.inaparadise.fragment.DialogBuatKomunitas

class IntroActivity : AppCompatActivity() {
    private var viewpager: ViewPager? = null
    private var liner: LinearLayout? = null
    private var myadapter: SliderAdapter? = null
    private var mdots: Array<TextView?>
    private var next: Button? = null
    private var back: Button? = null
    private var mulai: Button? = null
    private var mCureentPage = 0
    var viewlistener: OnPageChangeListener = object : OnPageChangeListener {
        override fun onPageScrolled(
            position: Int,
            positionOffset: Float,
            positionOffsetPixels: Int
        ) {
        }

        override fun onPageSelected(position: Int) {
            adddots(position)
            mCureentPage = position
            if (position == 0) {
                next!!.isEnabled = true
                back!!.isEnabled = false
                back!!.visibility = View.INVISIBLE
                next!!.text = "lanjut"
                back!!.text = ""
            } else if (position == mdots.size - 1) {
                next!!.isEnabled = true
                back!!.isEnabled = true
                back!!.visibility = View.VISIBLE
                next!!.text = ""
                back!!.text = "kembali"
                setMulai(true)
            } else {
                next!!.isEnabled = true
                back!!.isEnabled = true
                back!!.visibility = View.VISIBLE
                next!!.text = "lanjut"
                back!!.text = "kembali"
                setMulai(false)
            }
        }

        override fun onPageScrollStateChanged(state: Int) {}
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_intro)
        viewpager = findViewById<View>(R.id.viewpager) as ViewPager
        liner = findViewById<View>(R.id.dots) as LinearLayout
        next = findViewById<View>(R.id.nextBtn) as Button
        back = findViewById<View>(R.id.backBtn) as Button
        mulai = findViewById(R.id.mulai)
        myadapter = SliderAdapter(this)
        viewpager!!.adapter = myadapter
        adddots(0)
        viewpager!!.addOnPageChangeListener(viewlistener)
        next!!.setOnClickListener { viewpager!!.currentItem = mCureentPage + 1 }
        back!!.setOnClickListener { viewpager!!.currentItem = mCureentPage - 1 }
        mulai.setOnClickListener(View.OnClickListener {
            startActivity(
                Intent(
                    this@IntroActivity,
                    LoginActivity::class.java
                )
            )
        })
    }

    fun adddots(i: Int) {
        mdots = arrayOfNulls(4)
        liner!!.removeAllViews()
        for (x in mdots.indices) {
            mdots[x] = TextView(this)
            mdots[x]!!.text = Html.fromHtml("&#8226;")
            mdots[x]!!.textSize = 35f
            mdots[x]!!.setTextColor(resources.getColor(R.color.gray))
            liner!!.addView(mdots[x])
        }
        if (mdots.size > 0) {
            mdots[i]!!.setTextColor(resources.getColor(R.color.colorPrimary))
        }
    }

    fun setMulai(status: Boolean) {
        if (status) {
            mulai!!.visibility = View.VISIBLE
        } else {
            mulai!!.visibility = View.GONE
        }
    }
}